config.go

```diff
@@ -26,6 +28,41 @@ type defaultsHandler interface {
 	SetDefaults(c *Config) error
 }
 
+// Define backoffStrategy
+type BackoffStrategy interface {
+   Calculate(attempt int) time.Duration
+}
+
+type ExponentialStrategy struct {
+	Config *Config
+}
+
+func (s ExponentialStrategy) Calculate(attempt int) time.Duration {
+	backoffDuration := s.Config.BackoffMultiplier *
+		time.Duration(math.Pow(2, float64(attempt)))
+	if backoffDuration > s.Config.MaxBackoffDuration {
+		backoffDuration = s.Config.MaxBackoffDuration
+	}
+	return backoffDuration
+}
+
+// Full Jittered Backoff
+// http://www.awsarchitectureblog.com/2015/03/backoff.html
+type FullJitterStrategy struct {
+	Config *Config
+	Rng *rand.Rand
```

Comment:
my last question is, can we think of any way to avoid needing to instantiate these strategies with `Config` (and in this case, `Rng`)?

For `Rng`, we could use a private `sync.Once` to create it the first time `Calculate` is called.

Any ideas for `Config`?  I want it to be easy and obvious how to instantiate these properly *without* resorting to `NewXStrategy()` funcs.